### ISSUE TITLE
Origin or target Upload gives error on tsv/json files

### ISSUE THREAD

**Issue by monzug** [alpheios-project/alignment-editor-new#146]
@irina060981 the upload used to work but now I get errors for uploading a tsv file in both origin or target text. Could it be related to the fix to #135  and #133 ? see attachment
a minor thing, when I upload a file (well, I try to upload at the moment) the file name is sticky even after clear text. it's gone only after reloading the page. is it ok? what do you think?

![tsv-upload](https://user-images.githubusercontent.com/41396793/106610667-0b2b6f00-6535-11eb-889e-733507b360aa.JPG)

**Comment by monzug** [comment]
![json-upload-error](https://user-images.githubusercontent.com/41396793/106616417-8a23a600-653b-11eb-8419-f0031b82733e.JPG)

**Comment by irina060981** [comment]
>@irina060981 the upload used to work but now I get errors for uploading a tsv file in both origin or target text. Could it be related to the fix to #135 and #133 ? see attachment

yes it is for the fix - I set the limit for the txt and xml extension for the text single instance.
I could add csv, tsv - but they would be uploaded as is 

>a minor thing, when I upload a file (well, I try to upload at the moment) the file name is sticky even after clear text. it's gone only after reloading the page. is it ok? what do you think?

would fix it

**Comment by irina060981** [comment]
Merged and closed

**Comment by monzug** [comment]
@irina060981 the download of origin or target text does save the file as tsv, so for consistency the upload should work at least for this type of file too. I  think json too (let me think about this one). do you agree?

**Comment by irina060981** [comment]
>@irina060981 the download of origin or target text does save the file as tsv, so for consistency the upload should work at least for this type of file too. I think json too (let me think about this one). do you agree?

I will add tsv and csv - they would work as before
But we don't have an ability to upload a single text from json - because it was created only for global upload.
And we could not upload tokens only for one text - all texts should be aligned at once for now.

Other requirements would need more time for development.
@monzug , if you think that it is important to upload json with tokens for only one text then please - create a separate issue and we would need to define requirements for that there.

**Comment by monzug** [comment]
agree that there is not time for this but it used to work in this way: upload a json file in origin text, it worked only for the origin side - upload a json on target text, it uploaded only the target side. I was used to these work flow, not sure if it made any sense. I took it for granted. 
to recap: 
upload on top menu works for json, tsv, and ?
upload on origin/target text works with file types: txt, xml, csv, tsv (after this fix)

**Comment by irina060981** [comment]
>upload a json file in origin text, it worked only for the origin side - upload a json on target text, it uploaded only the target side. I was used to these work flow, not sure if it made any sense. I took it for granted.

It should not work - because you could download json only from the main menu.
In future we could return to these feature - but I would need more detailrd explanation about workflow here.


>to recap:
upload on top menu works for json, tsv, and ?
upload on origin/target text works with file types: txt, xml, csv, tsv (after this fix

Quite right - after the fix it would work this way - I didn't change the upload workflow - only set limits for extensions - to avoid uploading for example image files - as you showed me :)
I would expand limits for csv and tsv too - I forgot that this downloads as csv/tsv with params for direction, language and sourceType, I am sorry

**Comment by irina060981** [comment]
Merged and closed

**Comment by monzug** [comment]
Great. will do testing this aft and let you know. Thanks

**Comment by irina060981** [comment]
It is ready for testing :)

**Comment by monzug** [comment]
yes. upload for tsv and cvs works also in origin and target text. Thanks
